src/bindings/python/src/openvino/frontend/pytorch/fx_decoder.py

```diff
@@ -453,3 +453,104 @@ def mark_node(self, node):
         node.set_friendly_name(name)
         super().mark_node(node)
         return node
+
+      
+class ExecuTorchPythonDecoder (TorchFXPythonDecoder):
+
+    # TODO: The constructor of ExecuTorchPythonDecoder is mostly similar to the
+    # constructor TorchFXTorchPythonDecoder. Update this to utilize a common
+    # implementation.
+    def __init__(self, pt_module, fx_gm=None, nodes=None,
+                 mark_node_callback=None, input_shapes=[], input_types=[], dynamic_shapes=False):
+        super().__init__(mark_node_callback)
+        self.pt_module = pt_module
+        self.fx_gm = fx_gm if fx_gm is not None else pt_module
+        self.input_types = [OVAny(pt_to_ov_type_map[str(t)])
+                            for t in input_types]
+        self.input_shapes = input_shapes
+
+        self._input_signature = []
+        self._example_input = None
+
+        if issubclass(type(pt_module), torch.fx.graph_module.GraphModule):
+            self._input_is_list = None
+            self._nodes = list(pt_module.graph.nodes)
+            found_types = []
+            found_shapes = []
+            for i, value in enumerate(self._nodes):
+                if value.op == 'placeholder':
+                    self._inputs.append(i)
+                    self._input_signature.append(value.name)
+                    if hasattr(value, "meta") and ('tensor_meta' in value.meta.keys()) and value.meta['tensor_meta']:
+                        found_shapes.append(value.meta['tensor_meta'].shape)
+                        found_types.append(
+                            OVAny(pt_to_ov_type_map[str(value.meta['tensor_meta'].dtype)]))
+                    else:
+                        if hasattr(value, "meta") and ('val' in value.meta.keys()):
+                            found_shapes.append(value.meta["val"].shape)
+                            found_types.append(None)
+                        else:
+                            found_shapes.append(None)
+                            found_types.append(None)
+                elif value.op == 'output':
+                    # Instead of putting output index, refer to its target
+                    uargs = self.unpack_containers(value.args)
+                    self._outputs = [(arg[0], self._nodes.index(arg[1]))
+                                     for arg in uargs if arg[1] is not None]
+            for idx, shape in enumerate(found_shapes):
+                if shape is not None:
+                    new_shape = []
+                    for dim in shape:
+                        if (dynamic_shapes or type(dim).__name__ == "SymInt"):
+                            new_shape.append(-1)
+                        else:
+                            new_shape.append(dim)
+                    found_shapes[idx] = torch.Size(new_shape)
```

Comment:
I believe we should have helper function for this. Can you please check or create?
Why can't you reformat `found_shapes[idx]` right away when you were adding elements above? Now you have to re-iterate through it again

Reply:
It is possible to reformat when the elements are initially added into found_shapes. I updated accordingly.